### PR TITLE
[feat] Extract review heuristics into principle-code-review skill

### LINKED ISSUE #180: [feat] New skill: principle-code-review
## Goal

Extract the implicit review checklist from `agents/reviewer.md` into a standalone `principle-code-review` skill so other agents (`senior-engineer`, `auditor`, `architect`) can reference it without duplicating logic.

## Motivation

`agents/reviewer.md` encodes a rich set of review heuristics (correctness, security, design, test coverage, logic errors, security vulnerabilities) but these rules live only in that agent's system prompt. When `senior-engineer` or `auditor` need to perform review-like reasoning, they either duplicate the checklist or omit it. Extracting to a shared `principle-code-review` skill follows the same extraction pattern already applied to other cross-cutting concerns (testing, observability, security, performance).

Recent commits (`4797c84`, `ec42fc1`) show the project actively wiring principle skills into agents — this is the natural next extraction.

## Surface

- [x] Skill — `principle-*`
- [x] Subagent
- [x] Documentation

## Sketch

**Candidate principles to codify:**

- **Correctness first** — does the code do what it claims? Edge cases, off-by-ones, null/undefined paths.
- **Security lens** — OWASP Top 10 surface check: injection, auth bypass, insecure defaults, secret leakage, input validation at boundaries.
- **Design review** — is the abstraction at the right level? Does it introduce unnecessary coupling? Would a future reader understand it cold?
- **Test coverage** — are new behaviours covered? Are the tests behavioural (not implementation-bound)? Any coverage gaps in edge cases?
- **Confidence-based filtering** — only surface issues where confidence is high enough that fixing them is clearly worth the friction. Avoid nitpicks that don't improve correctness or safety.
- **Tone** — review the code, not the author. Phrase as observations, not accusations.

Once extracted, update `agents/reviewer.md` to load this skill rather than re-stating the rules inline.

## Acceptance

- [ ] `skills/principle-code-review.md` exists and follows the `principle-*` frontmatter schema.
- [ ] `agents/reviewer.md` updated to reference the new skill (deduplication).
- [ ] `python3 scripts/validate.py` passes.
- [ ] `docs/catalog.md` Skills table includes the new skill.
- [ ] At least one test covers the new skill file.

### PR DESCRIPTION
**Pull request by lugassawan**
## Summary
- Extract `## Focus`, `## Explicitly ignore`, and `## Judgement rules` from `agents/reviewer.md` into new `skills/principle-code-review/SKILL.md` with four-axis lens, confidence-based filtering, and tone guidance
- Wire the skill into `agents/reviewer.md` and `agents/auditor.md` via `## Principle consultation` bullets; `reviewer.md` gains a one-line pointer replacing the inline heuristics
- Add catalog entries to `agents/shared/skills.md` and `docs/catalog.md` (alphabetical between `principle-clean-code` and `principle-concurrency`)

## Test Plan
- [x] Changed skills/commands/agents load without errors (`python3 scripts/validate.py` → exits 0)
- [x] Examples in the diff were exercised manually (6 BM25 trigger fixtures all rank `principle-code-review` #1)
- [x] Full test suite: 331 passed

Closes #180

### CODE REVIEW

File: agents/reviewer.md [hunk 1]
```diff
@@ -7,16 +7,7 @@ tools: Read, Grep, Glob, Bash, Skill
...
-- **Correctness** — off-by-ones, null paths, concurrency races, lost errors, unhandled edge cases.
-- **Security** — injection, auth/authz gaps, secrets in code, unsafe deserialization, SSRF, missing input validation at trust boundaries.
-- **Design integrity** — SOLID violations, leaky abstractions, tight coupling, circular deps, domain logic bleeding into infrastructure.
-- **Tests** — missing coverage on new branches, brittle tests, tests that mirror implementation rather than behavior.
-
-## Explicitly ignore
-- Formatting, import order, quote style — that is the linter.
-- Stylistic preferences with no behavioral impact.
-- Speculative "could be" comments without a concrete failure mode.
+> Review heuristics (four-axis lens, confidence filtering, tone, what's not a finding): invoke `swe-workbench:principle-code-review`.
```
Comment (lugassawan) [inline agents/reviewer.md]: **High** — The blockquote pointer is advisory, not imperative. An agent can read line 10, treat it as informational, skip the Skill call, and start reviewing with no heuristics loaded (no confidence floors, no tone guidance, no nitpick filter).

Consider making it an explicit step 0 in `## Process`:

```md
## Process
0. **Load heuristics first.** Invoke `swe-workbench:principle-code-review` before reading the diff.
1. Read the diff end-to-end before commenting.
```

Or strengthen the blockquote phrasing:

```md
> Before reviewing any diff, invoke `swe-workbench:principle-code-review` to load
> the four-axis lens, confidence floors, tone rules, and nitpick filter.
```

File: skills/principle-code-review/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+---
+name: principle-code-review
```
Comment (lugassawan) [inline skills/principle-code-review/SKILL.md:2]: **Medium** — The acceptance criterion requires "at least one test covers the new skill file." The 6 BM25 trigger tests validate `description` token ranking but not the body content. If someone rewrites `SKILL.md`, nothing catches a regression in the heuristics (e.g. accidentally deleting the Confidence-Based Filtering section).

Suggest adding a structural smoke test to `tests/test_validate.py` (or a dedicated file):

```python
from pathlib import Path

class TestPrincipleCodeReviewSkill:
    SKILL_PATH = Path(__file__).parent.parent / "skills" / "principle-code-review" / "SKILL.md"

    def test_file_exists(self):
        assert self.SKILL_PATH.exists()

    def test_frontmatter_name(self):
        assert "name: principle-code-review" in self.SKILL_PATH.read_text()

    def test_four_axis_section_present(self):
        assert "## Four-Axis Review Lens" in self.SKILL_PATH.read_text()

    def test_confidence_filtering_section_present(self):
        assert "## Confidence-Based Filtering" in self.SKILL_PATH.read_text()
```

**Review by lugassawan — COMMENTED** [review]
Reviewed by `reviewer` (swe-workbench). Posted 2 inline comments, deduped 0.

**High** — `agents/reviewer.md:10`: blockquote pointer is advisory, not imperative; an agent can skip the Skill call and review with no heuristics loaded. Suggest making it step 0 in `## Process` or strengthening the blockquote to be explicitly directive.

**Medium** — No content-contract test for `SKILL.md` body; BM25 trigger tests validate description tokens but won't catch regression in the extracted heuristics (e.g. deleted sections). Suggest adding a structural smoke test.

The extraction itself is clean — the SKILL.md content is materially richer than the original bullets (confidence floor table, Red Flags table, "When Code Review Hurts"), trigger prompts are distinct, validate.py passes, and all 331 tests are green.
